core/common/assets/js/api/core/store.js

```diff
@@ -0,0 +1,125 @@
+import { configureStore, combineReducers } from '@reduxjs/toolkit';
+
+export default class Store {
+	/**
+	 * Initialize the Store.
+	 *
+	 * @return {void}
+	 */
+	constructor() {
+		this.slices = {};
+		this.reduxStore = this.createStore();
+	}
+
+	/**
+	 * Create a Redux Store object.
+	 *
+	 * @return {EnhancedStore<{}, AnyAction, [ThunkMiddlewareFor<{}>]>}
+	 */
+	createStore() {
+		const store = configureStore( {
+			// Use an empty function instead of empty object since an empty object
+			// isn't a valid reducer value.
+			reducer: () => {},
+		} );
+
+		// See: https://redux.js.org/usage/code-splitting#defining-an-injectreducer-function
+		store.asyncReducers = {};
+
+		store.injectReducer = ( key, newReducer ) => {
+			store.asyncReducers[ key ] = newReducer;
+
+			store.replaceReducer( combineReducers( { ...store.asyncReducers } ) );
+		};
+
+		return store;
+	}
+
+	/**
+	 * Register a Redux Store slice.
+	 *
+	 * @param {string} sliceId - Slice unique ID.
+	 * @param {Slice} instance - Slice object to add.
+	 *
+	 * @return {void}
+	 */
+	register( sliceId, instance ) {
+		if ( this.slices[ sliceId ] ) {
+			throw `Slice with ID '${ sliceId }' already exists.`;
+		}
+
+		this.slices[ sliceId ] = instance;
+
+		this.reduxStore.injectReducer( sliceId, instance.reducer );
```

Comment:
```suggestion
		const prevReducers = Object.fromEntries(
			Object.entries( this.slices )
				.map(([key, slice]) => [ key, slice.reducer ]))
		);

		this.reduxStore.replaceReducer(
			combineReducers( {
				...prevReducers,
				[sliceId]: instance.reducer
			} )
		)
```

Can replace injectReducer function. 
this logic can be handled inside another function

Reply:
Done